api/src/main/java/org/pmiops/workbench/notebooks/NotebookUtils.java

```diff
@@ -54,18 +64,43 @@ public static boolean isRStudioFile(String nameWithFileExtension) {
     return isRmdNotebook(nameWithFileExtension) || isRScriptFile(nameWithFileExtension);
   }
 
+  private static boolean isSasFile(String nameWithFileExtension) {
+    return nameWithFileExtension.endsWith(SAS_EXTENSION);
+  }
+
+  /** Appends file type extension e.g. ipynb if the file does not have one. */
+  public static String appendFileExtensionIfNotExist(
+      String fromNotebookNameWithExtension, String newName) {
+    if (isRmdNotebook(fromNotebookNameWithExtension)) {
```

Comment:
This doesn't include .R script files.  Unless we have a specific reason to exclude it (I don't think we do) then we should add it here as well.

Reply:
From what I remembered: our backend file sync (welder will sync R files into bucket). But we don't display R in analysis tab. (I forget the reason, maybe R is specific for RShiny app? Or we are at launch pressure about support RShiny)? 
But anyway, I feel it's a separate RStudio feature, we can check with product again and create separate ticket for that feature?

Reply:
ah, if you are just asking for support append suffix for R file. not general manage R file in analysis tab. I added R support in this method.

Reply:
We do show R files in the analysis tab.